v-next/hardhat/src/internal/builtin-plugins/network-manager/type-validation.ts

```diff
@@ -34,13 +120,77 @@ const httpNetworkUserConfigSchema = z.object({
   gas: z.optional(userGasSchema),
   gasMultiplier: z.optional(z.number()),
   gasPrice: z.optional(userGasSchema),
+  accounts: z.optional(httpNetworkUserConfigAccountsSchema),
 
   // HTTP network specific
   url: sensitiveUrlSchema,
   timeout: z.optional(z.number()),
   httpHeaders: z.optional(z.record(z.string())),
 });
 
+const keyBalanceObject = z.object({
+  privateKey: accountsPrivateKeySchema,
+  balance: z.string({ message: `${ACCOUNTS_ERROR} balance must be a string` }),
+});
+
+const canBeValidatedAsEdrKeyAndBalance = (item: unknown) => {
+  return isObject(item) && "privateKey" in item && "balance" in item;
+};
+
+const edrNetworkHDAccountsUserConfig = z.object({
+  mnemonic: z.optional(
+    z.string({
+      message: HD_ACCOUNT_MNEMONIC_MSG,
+    }),
+  ),
+  initialIndex: z.optional(
+    z
+      .number({
+        message: HD_ACCOUNT_INITIAL_INDEX_MSG,
+      })
+      .int({ message: HD_ACCOUNT_INITIAL_INDEX_MSG }),
+  ),
+  count: z.optional(
+    z
+      .number({
+        message: HD_ACCOUNT_COUNT_MSG,
+      })
+      .int({
+        message: HD_ACCOUNT_COUNT_MSG,
+      })
+      .positive({
+        message: HD_ACCOUNT_COUNT_MSG,
+      }),
+  ),
+  path: z.optional(
+    z.string({
+      message: HD_ACCOUNT_PATH_MSG,
+    }),
+  ),
+  accountsBalance: z.optional(
+    z.string({
+      message: HD_ACCOUNT_BALANCE_MSG,
+    }),
+  ),
+  passphrase: z.optional(
+    z.string({
+      message: HD_ACCOUNT_PASSPHRASE_MSG,
+    }),
+  ),
+});
+
+const edrNetworkUserConfigAccountsSchema = conditionalUnionType(
+  [
+    [
+      (data) =>
+        Array.isArray(data) && data.every(canBeValidatedAsEdrKeyAndBalance),
+      z.array(keyBalanceObject),
+    ],
+    [canBeValidatedAsHdAccount, edrNetworkHDAccountsUserConfig],
```

Comment:
tbh these checks may be too much. I'd simplify them as:

```suggestion
    [(data) => Array.isArray(data), z.array(keyBalanceObject)],
    [(data) => isObject(data), edrNetworkHDAccountsUserConfig],
```

but let's leave them as they are and get feedback from Pato after the alpha.